src/main/java/binaries/melissa/BinaryArray.java

```diff
@@ -2,22 +2,28 @@
 
 public class BinaryArray {
 
-    /**
-     * Array of 1s and 0s expected
-     */ 
-    private int[] binaries;
+	/**
+	 * Array of 1s and 0s expected
+	 */
+	private int[] binaries;
 
-    /**
-     * Simple constructor that assigns binaries list to instace's state
-     */
-    public BinaryArray(int[] binaries) {
-        this.binaries = binaries;
-    }
+	/**
+	 * Simple constructor that assigns binaries list to instace's state
+	 */
+	public BinaryArray(int[] binaries) {
+		this.binaries = binaries;
+	}
 
-    /**
-     * Returns the decimal numeric value equivalent to the list of binaries stored
-     */
-    public int ConvertToDecimalInt() {
-        throw new UnsupportedOperationException("Not supported yet.");
-    }
+	/**
+	 * Returns the decimal numeric value equivalent to the list of binaries stored
+	 */
+	public int ConvertToDecimalInt() {
+		int decimal = 0;
+		double powOfTwo;
+		for (int i = 0; i < binaries.length; i++) {
```

Comment:
Could we use a more meaningful variable name instead of `i`?

Reply:
Changed name to "BASE_NUMBER" as a constant

Reply:
Changed to "index"